.rhtap/scripts/subscription-manager-bro.sh

```diff
@@ -0,0 +1,218 @@
+    "diff")
+        fn=assert_diff ;;
+    *)
+        >&2 echo "Error: unknown command '$1'; call '$SCRIPT_NAME help' to see usage."
+        exit 3
+        ;;
+    esac
+
+    if [[ "$#" -gt "1" && "$cmd" != "diff" ]]; then
+        >&2 echo "Error: too many arguments; call '$SCRIPT_NAME help' to see usage."
+        exit 4
+    fi
+
+    $fn "$@"
+}
+
+function usage {
+    echo "Usage: $SCRIPT_NAME smuggle|register|cleanup|self-test"
+    echo
+    echo "This script enables access to RHEL RPMs during RHTAP builds. The intended usage is as follows."
+    echo
+
+    echo -n "1. Make sure there is '$RHTAP_SECRET_NAME' secret in RHTAP with key name '$SECRET_KEY' and "
+    echo "the actual activation key as a value."
+
+    echo -n "2. In a Tekton pipeline step before the container build, copy the subscription manager activation "
+    echo "key secret to the source workspace. Use:"
+    echo "   \$ <source-workspace>/$SCRIPT_NAME smuggle"
+    echo -n "   This expects '$RHTAP_SECRET_NAME' secret to be mounted as a workspace with the same name "
+    echo "('$RHTAP_SECRET_NAME')."
+
+    echo "3. Arrange Dockerfile stages so that there's UBI (normal) as installer and other RHEL/UBI (any) as target."
+    echo "   Copy target contents to ${TARGET_DIR} in the installer stage. See self-test Dockerfiles as examples."
+
+    echo "4. In installer stage, register the container with subscription manager. Use:"
+    echo "   \$ $SCRIPT_NAME register"
+
+    echo -n "5. Use 'dnf --installroot=${TARGET_DIR} ...' to install RHEL RPMs, enable RHEL modules, etc. "
+    echo "in target contents."
+
+    echo -n "6. In the same installer stage, deregister the container so that end users can't use "
+    echo "our subscription on our behalf. Use:"
+    echo "   \$ $SCRIPT_NAME cleanup"
+    echo "   This step is mandatory because it cleans entitlements on target in the right way."
+
+    echo -n "7. Copy out ${TARGET_DIR} contents from the installer stage to a new 'scratch' stage. "
+    echo "That's your target container."
+
+    echo
+    echo -n "If you need to install entitled RPMs into multiple stages in one Dockerfile, use separate installer stage for each. "
+    echo "This is because the script assumes one distinct system to be in ${TARGET_DIR}."
+    # I did not bother to make it more universal since it's already a lot for a throw-away.
+
+    echo
+    echo "When altering this script, use self-test command as (regression) test tool:"
+    echo "   \$ $SCRIPT_NAME self-test"
+    echo "For it to work, you need to put a valid activation key in ${SECRET_PATH} file."
+}
+
+function smuggle {
+    mkdir -p "${REPO_ROOT}/.rhtap/build/"
+    cp --verbose "/workspace/${RHTAP_SECRET_NAME}/${SECRET_KEY}" "${SECRET_PATH}"
+}
+
+function register {
+    if [[ ! -d "${TARGET_DIR}"/etc ]]; then
+        >&2 echo "Error: Looks like target system is not mounted at ${TARGET_DIR}/etc"
+        exit 5
+    fi
+
+    echo "Backing up original artifacts in $TARGET_DIR"
```

Comment:
Why is this necessary?

Reply:
ok, after reading on I see that they will be restored in the cleanup function. A comment would be nice.

Reply:
Added.